pkg/machine/ocipull/ociartifact.go

```diff
@@ -92,8 +109,17 @@ func NewOCIArtifactPull(ctx context.Context, dirs *define.MachineDirs, endpoint
 
 	cache := false
 	if endpoint == "" {
-		imageName := artifactImageName
-		endpoint = fmt.Sprintf("docker://%s/%s/%s:%s", artifactRegistry, artifactRepo, imageName, artifactVersion.majorMinor())
+		// Get machine image from configuration
+		machineImage, err := getMachineImageConfig()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get machine image configuration: %w", err)
+		}
+
+		if _, err := validateMachineImageConfig(machineImage); err != nil {
+			return nil, fmt.Errorf("machine image ref could be misconfigured in containers.conf: %w", err)
+		}
+
+		endpoint = fmt.Sprintf("docker://%s:%s", machineImage, artifactVersion.majorMinor())
```

Comment:
this would not be right, the config filed could also be a normal http url or local file path see my comment on the common PR, the docker:// prefix must be part of the config field.